docs/infra/spk-infra-generation-pipeline.md

```diff
@@ -0,0 +1,115 @@
+# Guide: Infrastructure Generation Pipeline
```

Comment:
@yradsmikham there are several steps where we are asking folks to perform manual actions. We have commands in `spk` that will help with creating pipelines, variable groups etc.

Is there a reason we are not guiding people to use `spk`?

Reply:
Just reached out to folks on the services side, it seems like I might be able to use `spk` to generate variables groups, but maybe not creating pipelines. We can fall back on the `az cli` to automate pipeline creation for now. Will make these changes in the doc! Thanks!